File: Step 03 Arrays/Step 3.3 Hard/LeetCode/FindRepeating&Missing.java
import java.util.HashMap;
import java.util.Arrays;

public class Solution {
    public static int[] findMissingRepeatingNumbers(int []a) {
        // Write your code here
        HashMap<Integer, Boolean> hMap = new HashMap<>();
        int repeatedOne = 0;
        int messedSum = Arrays.stream(a).sum();
        for(int x : a) {
            if(hMap.containsKey(x)) {
                repeatedOne = x;
                break;
            } else {
                hMap.put(x, true);
            }
        }
        int n = a.length;
        int originalSum = (n*(n+1)) / 2;
        return new int[]{repeatedOne, Math.abs(originalSum - (messedSum - repeatedOne))};
    }
}

File: Step 03 Arrays/Step 3.3 Hard/FindRepeating&Missing/FindRepeating&Missing.cpp
class Solution{
public:
    vector<int> findTwoElement(vector<int> arr, int n) {
        // code here
        vector<int> hash(n);
        vector<int> ans(2);
        for(int i: arr){
            hash[i-1]++;
        }
        for(int i = 0; i < n; i++){
            if(hash[i] == 2){
                ans[0] = i+1 ;
            }
            if(hash[i] == 0){
                ans[1] = i+1;
            }
        }
        return ans;
    }
};